### PR TITLE
Add more tests for WebHeaderCollection

### PR DESCRIPTION
**Pull request by Maxwe11**
Excluded from #2380

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @CIPop @stephentoub @SidharthNabar

**Comment by stephentoub** [comment]
LGTM

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs [hunk 1]
```diff
@@ -283,5 +293,98 @@ public void IterateIEnumerable_Success()
...
+            w.Add("Accept", "text/plain");
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => w.GetKey(-1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => w.GetKey(42));
+        }
+
+        [Fact]
+        public void Clear_Success()
+        {
+            var w = new WebHeaderCollection();
```
Comment (davidsh) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: @Maxwe11 In the rest of the new code, you have this 'WebHeaderCollection w = new WebHeaderCollection();` but here you use a `var`.  So, please be consistent.  You can use the `var` on all the cctor's or none of them.

Comment (Maxwe11) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: fixed

**Comment by davidsh** [comment]
LGTM

**Pull request merged**